Source/Core/InputCommon/ControlReference/ExpressionParser.cpp

```diff
@@ -488,6 +510,32 @@ class UnarySinExpression : public UnaryExpression
   std::string GetFuncName() const override { return "Sin"; }
 };
 
+class UnaryWhileExpression : public UnaryExpression
+{
+public:
+  UnaryWhileExpression(std::unique_ptr<Expression>&& inner_) : UnaryExpression(std::move(inner_)) {}
+
+  ControlState GetValue() const override
+  {
+    constexpr int MAX_REPS = 10000;
+    constexpr int COND_THRESHOLD = 0.5;
+
+    // Returns 1.0 on successful loop, 0.0 on reps exceeded. Sensible?
+
+    for (int i = 0; i != MAX_REPS; ++i)
+    {
+      const ControlState val = inner->GetValue();
+      if (val < COND_THRESHOLD)
+        return 1.0;
+    }
+
+    // Exceeded max reps:
+    return 0.0;
+  }
+  void SetValue(ControlState value) override {}
+  std::string GetFuncName() const override { return "Sin"; }
```

Comment:
I don't think your loop should be called `Sin` ;)

Reply:
Nice catch. ;)